openid/idtokenvalidator.go

```diff
@@ -60,15 +64,23 @@ func (tv *idTokenValidator) renewAndGetSigningKey(jt *jwt.Token) (interface{}, e
 	if err != nil {
 		return nil, err
 	}
+	kid, ok := jt.Header[keyIDJwtHeaderName].(string)
```

Comment:
Glad you found this issue, which reminds me that per openid spec the kid is optional. So it should be fine to use getTokenKid inside both getSigningKey and renewAndGet but with a different implementation... getTokenKid should not return an error when no kid is found, instead it should return "". This should also take care of the failed test.